### DISCUSSION TITLE
Fatslane Setting up iOS notification extensions

### DISCUSSION THREAD

**Discussion by aryan-ghassemi-ck** [fastlane/fastlane#19152] [Q&A]
I have created a notification extension.
My main app id is `com.company.app`
My notification extension id is `com.company.app.notificationService`

I have created a new app id on iOS dev portal with identifier `com.company.app.notificationService` and enabled Associated Domain and Push Notifications

I call the following to have match generate a new provisioning profile.
It creates a new provisioning profile with app id as the new identifier I created in the previous step

```
match(
   git_url: "https://github.com/company/certs.git",
   type: "enterprise",
   app_identifier: ["com.company.app.notificationService"],
   readonly: !is_ci
)
```

I have added logic to apply provisioning
```
// filtered to ensure we only change the provisioning profile of the extension
update_project_provisioning(
   xcodeproj: ENV["PROJECT"],
   target_filter: ENV["com.company.app.notificationService"],
   profile: ENV["sigh_com.company.app.notificationService_enterprise_profile-path"],
   build_configuration: "Debug"
)
```

Running this I get the following error. It looks like it's comparing the provisioning profile with the main app and not with the extension, although I have added my extension target as a filter
```
error: Provisioning profile "match InHouse com.company.app.notificationService" has app ID "com.company.app.notificationService", which does not match the bundle ID "com.company.app". (in target 'Main App' from project 'Project')

```

**Comment by lucgrabowski** [comment]
Hi @aryan-ghassemi-ck , did you set `com.company.app.notificationService` environment variable?
There is an example of how to use `target_filter` in docs:
http://docs.fastlane.tools/actions/update_project_provisioning/#update_project_provisioning
`target_filter: ".*WatchKit Extension.*", # matches name or type of a target`
In the first step you could hardcode `target_filter` value in the call without using environment variables to see how to use it.

**Reply by aryan-ghassemi-ck** [reply]
Hi yes, I have prints and all environment variables seem to be there

**Comment by John9656** [comment]
I have also same error any solution for that.

[RapidFS](https://www.rapidfs.biz/)

**Comment by lucgrabowski** [comment]
Hi, I spent time investigating logs that you provided in https://github.com/fastlane/fastlane/issues/19231

> [10:18:45]: Update App porvisioning
[10:18:45]: Target Filter: CompanyUniversal
...
[10:18:47]: Update NotificationService porvisioning
[10:18:47]: Target Filter:


It looks like environment variable used for extension's`target_filter` `ENV["com.company.app.notificationService"]` is not defined.
That's why `update_project_provisioning` updated provisionings for all the targets (take a look at logs below `[10:18:45]: --- Step: update_project_provisioning ---`) and you got a mismatch in main app target between main app bundle identifier and provisioning profile that was meant for NotificationService extension.
`❌  error: Provisioning profile "match InHouse com.company.beta.NotificationService" has app ID "com.company.beta.NotificationService", which does not match the bundle ID "com.company.beta". (in target 'companyUniversal' from project 'companyUniversal')`

You can confirm it by opening xcode and taking a look at signing that was set in the project.
To fix the error you can set variable `com.company.app.notificationService`
or hardcode:
```
    update_project_provisioning(
      xcodeproj: ENV["CK_PROJECT"],
      target_filter: "NotificationService",
      profile: ENV["sigh_com.company.beta.NotificationService_enterprise_profile-path"],
      build_configuration: "Debug"
    )
```
